TradierClient/TradierClient/TradierStreamingClient.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Tradier.Client.Models.Streaming;
+
+// ReSharper disable once CheckNamespace
+namespace Tradier.Client
+{
+    /// <summary>
+    /// The <c>Streaming</c> class
+    /// </summary>
+    public class TradierStreamingClient : IDisposable
+    {
+        private readonly string _uri;
+        private readonly System.IO.Stream _stream;
+        private readonly StreamReader _streamReader;
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+
+        /// <summary>
+        /// The Streaming constructor
+        /// </summary>
+        public TradierStreamingClient(string endpoint, string content = null)
+        {
+            _jsonSerializerSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore
+            };
+
+            _uri = endpoint;
+            var request = (HttpWebRequest)WebRequest.Create(_uri);
+
+            request.Accept = "application/json";
+
+            if (content == null)
+            {
+                request.Method = HttpMethod.Get.ToString().ToUpper();
+            }
+            else
+            {
+                request.ContentType = "application/x-www-form-urlencoded";
+                request.Method = HttpMethod.Post.ToString().ToUpper();
+                var data = Encoding.ASCII.GetBytes(content);
+                request.ContentLength = data.Length;
+                using (var stream = request.GetRequestStream())
+                {
+                    stream.Write(data, 0, data.Length);
+                }
+            }
+
+            var response = (HttpWebResponse)request.GetResponse();
+
+            _stream = response.GetResponseStream();
+            _streamReader = new StreamReader(_stream);
+
+            onOpened(this, new StateChangedEventArgs(ReadyState.Open));
+        }
+
+        public async Task StartAsync()
+        {
+            while (!_streamReader.EndOfStream)
```

Comment:
I feel like this part is sort of janky. I had some trouble with using the approach that was outlined in Tradier's documentation. Specifically, using the _streamReader.Readline method just hung whenever it was called. I think it might have something to do with the way the data is formatted as it is coming in and there is maybe not an EOL character or something like that. I am not super familiar with how this is supposed to work though. At any rate, I came up with this approach, which does seem to work fine; however, I did see a couple of times where the currentLine (below) was cut off and the subsequent Json Serialization didn't work as a result. I did make a couple of changes to the way I was retrieving the streamed data, and I think it might have fixed that issue, but it is a bit hard to test.